packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -100,6 +100,13 @@ function isInsideComponentOrHook(node) {
   return false;
 }
 
+function isUseEventIdentifier(node) {
+  return (
+    node.type === 'Identifier' &&
+    (node.name === 'useEvent' || node.name === 'experimental_useEvent')
```

Comment:
It seems like it shouldn't be called `experimental_useEvent` in code anyway because it wouldn't be considered a Hook then. So maybe let's assume it's always aliased ot `useEvent`.

I propose that we only check `useEvent` here and wrap this check in `__EXPERIMENTAL__`. If `__EXPERIMENTAL__` is `false`, always return `false` from this function.

Then let's copy and paste this in the other file so that detection is done the same way. E.g. the other file doesn't currently check the flag. The other file also doesn't consider `useEvent` as stable but it should (under the flag, that is).